src/test/java/com/williamfiset/algorithms/search/QuickSelectTest.java

```diff
@@ -0,0 +1,17 @@
+package com.williamfiset.algorithms.search;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.Test;
+
+public class QuickSelectTest {
+
```

Comment:
Can you add a test which fully tests the functionality of your quickselect implementation? I'm thinking of something like:

1. Generate a range array of 100 random ints
1. Make a sorted copy of the generated array
1. Use the quick select algorithm to find the k'th smallest elements for all values of k between 1 and n.
1. Make sure the quick select algorithm found the k'th smallest element by checking the k'th element in the sorted array